truffle/com.oracle.truffle.object.dsl.processor/src/com/oracle/truffle/object/dsl/processor/LayoutGenerator.java

```diff
@@ -0,0 +1,901 @@
+                stream.println("            super(");
+
+                iterateProperties(layout.getInheritedShapeProperties(), new PropertyIteratorAction() {
+
+                    @Override
+                    public void run(PropertyModel property, boolean last) {
+                        stream.printf("                %s", property.getName());
+
+                        if (last) {
+                            stream.println(");");
+                        } else {
+                            stream.println(",");
+                        }
+                    }
+
+                });
+            }
+
+            iterateProperties(layout.getShapeProperties(), new PropertyIteratorAction() {
+
+                @Override
+                public void run(PropertyModel property, boolean last) {
+                    stream.printf("            this.%s = %s;%n", property.getName(), property.getName());
+                }
+
+            });
+
+            stream.println("        }");
+            stream.println("        ");
+
+            for (PropertyModel property : layout.getAllShapeProperties()) {
+                final boolean inherited = !layout.getShapeProperties().contains(property);
+
+                if (!inherited) {
+                    stream.printf("        public %s %s() {%n", property.getType(), NameUtils.asGetter(property.getName()));
+                    stream.printf("            return %s;%n", property.getName());
+                    stream.println("        }");
+                    stream.println("        ");
+                }
+
+                if (inherited) {
+                    stream.println("        @Override");
+                }
+
+                stream.printf("        public %sType %s(%s %s) {%n", layout.getName(), NameUtils.asSetter(property.getName()), property.getType(), property.getName());
+                stream.printf("            return new %sType(%n", layout.getName());
+
+                iterateProperties(layout.getAllShapeProperties(), new PropertyIteratorAction() {
+
+                    @Override
+                    public void run(PropertyModel p, boolean last) {
+                        stream.printf("                %s", p.getName());
+
+                        if (last) {
+                            stream.println(");");
+                        } else {
+                            stream.println(",");
+                        }
+                    }
+
+                });
+
+                stream.println("        }");
+                stream.println("        ");
+            }
+        } else {
+            stream.println("        ");
+        }
+
+        stream.println("    }");
+        stream.println("    ");
+    }
+
+    private void generateAllocator(final PrintStream stream) {
+        if (layout.getSuperLayout() == null) {
+            stream.println("    protected static final Layout LAYOUT = Layout.newLayout().addAllowedImplicitCast(Layout.ImplicitCast.IntToLong).build();");
```

Comment:
`addAllowedImplicitCast(Layout.ImplicitCast.IntToLong)`
is Ruby specific isn't it? Is there a way to configure this? There should be.